lib/Database/lmdb/mdb.c

```diff
@@ -164,7 +164,7 @@ typedef SSIZE_T	ssize_t;
 # define MDB_USE_SYSV_SEM	1
 # endif
 # define MDB_FDATASYNC		fsync
-#elif defined(ANDROID)
+#elif defined(__ANDROID__)
```

Comment:
You need the underscores, [as can be seen elsewhere in this repo](https://github.com/apple/indexstore-db/blob/master/lib/LLVMSupport/Support/Unix/Threading.inc#L96).